### DISCUSSION TITLE
Assignment 1 - Poker Problem - Straight

### DISCUSSION THREAD

**Discussion by rockyx50** [chulminy/CIVE497-CIVE700#112] [Q&A]
Hi there,

I saw this line of code for converting a logical vector into a character array in the solution provided. I was wondering if this was the standard way of doing this or is there another method?

`is_str = contains(char(lg_vec + '0'), '11111');`

I copied this code into my local matlab and followed the debugging steps to understand it but I was not able to replicate what this line of code does in another way . From what I gathered the + '0' converts the logical vector into a double type vector and the char turns it into a character array of the ASCII values.

**Comment by chulminy** [comment]
Good question.  
Your explanation is correct. 
This is the simplest way to convert a logical vector to a character vector. 
I'm not sure this is a formal way. Are they any suggestion? 

How about these?

`sprintf('%d', lg_vec)`

or 

`char(double(lg_vec) + double('0'))`

or

`num2str(lg_vec,'%d')`

For me, I like mine but you can choose one of these.

**Reply by rockyx50** [reply]
Thanks! I also notice for the second option you add a '0' at the end of the double  logical vector. Do character arrays need to end in 0?

**Reply by chulminy** [reply]
This is the process to create numeric values corresponding to '0' or '1'. 

https://youtu.be/2TDD6vV6Lfg?list=PLa1nAPP8qUX90fanmEze3zLD41N-4lV25&t=1708

This might be helpful.

**Comment by zaalsabb** [comment]
Here's an answer that I saw on MATLAB answers for converting logical arrays to doubles arrays, which just uses the double() function:
https://www.mathworks.com/matlabcentral/answers/388474-convert-logical-vector-to-double#answer_310172

```
a = [true false];
double(a)
```

But in your question, you want to convert a logicals array to a strings array, so you would wrap it in a string() function to convert the logicals array to a doubles array, then to a strings array.

```
string(double(a))
```

**Reply by chulminy** [reply]
No string array. The code above is to convert it into a character array. 

String and character are different. 

Also, the answer above is to convert it to a string array, not a numeric array. 

Character array is considered as a numeric array when you involve a numeric operation. 

This is the way of using `string`

![image](https://user-images.githubusercontent.com/3227103/105203845-91958900-5b11-11eb-926d-0fb769086169.png)

Does it make sense?

**Reply by zaalsabb** [reply]
Yes, for character arrays there is an extra step to convert the string vector. You can also use the char() function to convert a string, but you have to join the characters in the string vector first to a single string with no spaces in between. 
```
>> a = logical([1 0 1 0]);
>> b = string(double(a));
>> c = char(join(b,''))

c =

    '1010'

>> 
```
